core/frontend/src/ViewState.ts

```diff
@@ -189,6 +191,8 @@ export abstract class ViewState extends ElementState {
   private _categorySelector: CategorySelectorState;
   private _displayStyle: DisplayStyleState;
   private readonly _unregisterCategorySelectorListeners: VoidFunction[] = [];
+  protected _hydrateRequest: HydrateViewStateRequestProps = {};
```

Comment:
Storing this seemingly-temporary hydration state on the ViewState seems very weird to me. Did you consider passing arguments to preload and postload instead?

Reply:
Ah sorry.. I had decided it was necessary before I made it to the current approach. Changed it.